File: examples/elixir/user-server/test/support/user_service_server_probe.ex
defmodule UserServiceServerProbe do
  @moduledoc false

  @spec handle_get_user(term(), UserServiceTypes.GetUserInput.t(), term()) ::
          {:ok, UserServiceTypes.GetUserOutput.t()} | {:error, term()}
  def handle_get_user(_ctx, input, _meta) do
    {:ok, %UserServiceTypes.GetUserOutput{user: %{"userId" => input.user_id}}}
  end
end

File: examples/erlang/user-server/test/user_service_server_probe.erl
-module(user_service_server_probe).

-include("user_service_types.hrl").

-export([handle_get_user/3]).

%% Test double: partial handler module for router tests (not a behaviour implementor).
handle_get_user(_Ctx, #get_user_input{user_id = UserId}, _Meta) ->
    {ok, #get_user_output{
        user = #{<<"userId">> => UserId}
    }}.
